stock_picking_supplier_ref/models/stock_picking.py

```diff
@@ -0,0 +1,10 @@
+# Copyright 2017 Trey, Kilobytes de Soluciones
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    reference = fields.Char(string="Supplier Reference")
```

Comment:
IMHO reference is too more "open" definition. I mean, are we talking about the reference of the supplier_reference available in the PO, are we talking about the picking reference of the supplier ?
I would recommend to make some small changes to better specify the use case.